src/built-in-types/MapType.ts

```diff
@@ -5,27 +5,42 @@ export interface Spec {
   [key: string]: EcsType
 }
 
-export type Result<T extends Spec> = {
+/** Include property keys from T where the property is assignable to U */
+type IncludeUndefined<T> = {
```

Comment:
can we move all this types to a types.ts file ?

Reply:
done